src/components/LoginForm.component.tsx

```diff
@@ -0,0 +1,124 @@
+import { useLazyQuery } from "@apollo/client";
+import React, { useContext, useEffect, useState } from "react";
+import { useNavigate } from "react-router-dom";
+import styled from "styled-components";
+
+import { UserContext } from "../contexts/UserContext";
+import { VALIDATE_EMAIL } from "../graphql/queries";
+import { useLocalStorage } from "../hooks/useLocalStorage";
+import { EmailVars, Response } from "../interfaces/User";
+import { GlobalStyle } from "../theme";
+import { REPLACE, STORAGE_KEY } from "../utils/constants";
+import { Button, Form, H1, Input, ValidationText } from "./styled";
+
+function isValidEmail(email: string) {
+  const regex =
+    /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
+  return regex.test(email);
+}
+
+export const LoginForm = () => {
+  const { loggedUser, setLoggedUser } = useContext(UserContext);
+  const [errorMessage, setErrorMessage] = useState("");
+  const [emailInput, setEmailInput] = useState("");
+  const [, setUserInStorage] = useLocalStorage(STORAGE_KEY.USER, "");
+  const [validateEmail, { loading, error, data }] = useLazyQuery<
+    Response,
+    EmailVars
+  >(VALIDATE_EMAIL);
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    if (loggedUser) navigate("/", REPLACE);
```

Comment:
why are using this `REPLACE` on all the navigate functions?

Reply:
`REPLACE` is just an object `{ replace : true }`. I'm only using it to prevent that object to split into 3 different lines, I thought it could add readability, and it does for me, what do you think about it?